teps/0086-changing-the-way-result-parameters-are-stored.md

```diff
@@ -258,7 +260,7 @@ tasks:
   taskSpec: foo
   results:
   - name: out
-    byRef: true # <-- *new field*
+    type: stringReference # <-- *new type*
 - name: second-task
   taskSpec: bar
   params:
```

Comment:
The result `type` was introduced 0.36 as a alpha feature to identify whether a result is a `string` or an `array`. This  option is utilizing the same type and extending it to support references.

Reply:
stringReference/arrayReference or just "reference" as new type ? 

When the type is reference, shall we allow it to be passed as params? If we do allow, do we expand the references ?

Reply:
In the future there will be a new way for param to consume result as `array`. Having type stringReference just to easier to indicate the result is string and it's stored as a reference. We should allow it to be passed as params as the current results.